File: tools/cpp/tests/metrics/test_StreamingMAD.cc
// unittest_streaming_mad

#include <gtest/gtest.h>
#include "StreamingMAD.h"
#include <vector>

using KML::Metrics::StreamingMAD;

class StreamingMADTest : public ::testing::Test
{
    public:
        StreamingMAD* sm;

        StreamingMADTest()
        {
            sm = new StreamingMAD();
        }
};

TEST_F(StreamingMADTest, TestEmptyEvaluate)
{
    EXPECT_FLOAT_EQ(0.0, sm->evaluate());
}

TEST_F(StreamingMADTest, TestMADCalculationSimple1)
{
    std::vector<double> data{10, 10};
    sm->update(data);
    EXPECT_FLOAT_EQ(0.0, sm->evaluate());
}

TEST_F(StreamingMADTest, TestMADCalculationSimple2)
{
    std::vector<double> data{2, 1, 1};
    sm->update(data);
    EXPECT_FLOAT_EQ(0.0, sm->evaluate());
}

File: tools/python/tests/metrics/test_StreamingMAD.py
import unittest

from kml.metrics.StreamingMAD import PyStreamingMAD


class TestStreamingMAD(unittest.TestCase):

    DATA = list(range(1, 11))
    WINDOW_SIZE = len(DATA)
    MAD = None

    def setUp(self):
        self.MAD = PyStreamingMAD()

    def tearDown(self):
        self.MAD = None

    def test_update(self):
        self.MAD.update(self.DATA)

    def test_empty_evaluate(self):
        self.assertEqual(self.MAD.evaluate(), 0.0)
